fivetran_provider_async/hooks.py

```diff
@@ -257,18 +257,26 @@ def get_destinations(self, group_id: str) -> dict:
         resp = self._do_api_call("GET", endpoint)
         return resp["data"]
 
-    def get_groups(self, group_id: str) -> dict:
+    def get_groups(self, group_id: str = "") -> dict:
         """
-        Fetches destination information for the given group.
-        :param group_id: The Fivetran group ID, returned by a connector API call.
+        Fetches information about groups.
+        :param group_id: The Fivetran group ID, if provided will restrict to that group.
         :return: group details
         """
-        if group_id == "":
-            raise ValueError("No value specified for connector_id")
         endpoint = self.api_path_groups + group_id
         resp = self._do_api_call("GET", endpoint)
         return resp["data"]
 
+    def get_connectors(self, group_id: str) -> dict:
+        """
+        Fetches connector information for the given group.
+        :param group_id: The Fivetran group ID, returned by a connector API call.
+        :return: connector details
+        """
+        endpoint = f"{self.api_path_groups}{group_id}/connectors"
+        resp = self._do_api_call("GET", endpoint)
+        return resp["data"]
```

Comment:
can you please add a test for this?

Reply:
I added some tests, the rest responses are copied from https://fivetran.com/docs/rest-api/groups.
I realized since that the groups returns a paginated response, so I modified the original PR so it returns a generator if somebody calls `get_groups` without an argument, and a single item if somebody calls it with one.  Really I would have liked to add a `get_group` method, but at least this way the API is maintained.